perfmetrics/scripts/continuous_test/ml_tests/run_and_manage_test.sh

```diff
@@ -0,0 +1,192 @@
+TEST_SCRIPT_PATH=$4
+
+function delete_existing_vm_and_create_new () {
+  (
+    set +e
+    echo "Deleting VM $VM_NAME in zone $ZONE_NAME."
+    sudo gcloud compute instances delete $VM_NAME --zone $ZONE_NAME --quiet
+    if [ $? -eq 0 ];
+    then
+      echo "Machine deleted successfully !"
+    else
+      echo "Machine was not deleted as it doesn't exist."
+    fi
+  )
+
+  echo "Wait for 30 seconds for old VM to be deleted"
+  sleep 30s
+
+  echo "Creating VM $VM_NAME in zone $ZONE_NAME"
+  sudo gcloud compute instances create $VM_NAME \
+      --project=$GCP_PROJECT \
+      --zone=$ZONE_NAME \
+      --machine-type=a2-highgpu-2g \
+      --network-interface=network-tier=PREMIUM,nic-type=GVNIC,stack-type=IPV4_ONLY,subnet=default \
+      --metadata=enable-oslogin=true \
+      --maintenance-policy=TERMINATE \
+      --provisioning-model=STANDARD \
+      --service-account=927584127901-compute@developer.gserviceaccount.com \
+      --scopes=https://www.googleapis.com/auth/cloud-platform \
+      --accelerator=count=2,type=nvidia-tesla-a100 \
+      --create-disk=auto-delete=yes,boot=yes,device-name=$VM_NAME,image=projects/ubuntu-os-cloud/global/images/ubuntu-2004-focal-v20230523,mode=rw,size=200,type=projects/$GCP_PROJECT/zones/us-central1-c/diskTypes/pd-balanced \
+      --no-shielded-secure-boot \
+      --shielded-vtpm \
+      --shielded-integrity-monitoring \
+      --labels=goog-ec-src=vm_add-gcloud \
+      --reservation-affinity=any
+
+  echo "Wait for 30 seconds for new VM to be initialised"
+  sleep 30s
+
+  echo "Delete existing ssh keys "
+  # This is required to avoid issue: https://github.com/kyma-project/test-infra/issues/93
+  for i in $(sudo gcloud compute os-login ssh-keys list | grep -v FINGERPRINT); do sudo gcloud compute os-login ssh-keys remove --key $i; done
+
+  # Requires running first ssh command with --quiet option to initialize keys.
+  # Otherwise it prompts for yes and no.
+  sudo gcloud compute ssh $VM_NAME --zone $ZONE_NAME --internal-ip --quiet --command "echo 'Running from VM'"
+}
+
+# Takes commit id of on-going test run ($1) artifacts to GCS bucket
+function copy_run_artifacts_to_gcs () {
+  (
+    # We don't want to exit if failure occurs while copying GCSFuse logs because
+    # gsutil always gives error (even the files are copied) while uploading
+    # files that are changing while uploading and gcsfuse logs are changing
+    # when the test is running.
+    set +e
+    echo "Copying GCSFuse logs to GCS bucket"
+    sudo gcloud compute ssh $VM_NAME --zone $ZONE_NAME --internal-ip --command "gsutil rsync -R -d \$HOME/github/gcsfuse/container_artifacts/ $ARTIFACTS_BUCKET_PATH/$1/container_artifacts"
+    echo "\n"
+  )
+  sudo gcloud compute ssh $VM_NAME --zone $ZONE_NAME --internal-ip --command "gsutil cp \$HOME/build.out $ARTIFACTS_BUCKET_PATH/$1/build.out"
+  sudo gcloud compute ssh $VM_NAME --zone $ZONE_NAME --internal-ip --command "gsutil cp \$HOME/build.err $ARTIFACTS_BUCKET_PATH/$1/build.err"
```

Comment:
why not use kokoro artifacts functionality instead of manually copying the files?

Reply:
got it. may be its not viable since every time you try to check the status it will create a seperate folder.

Reply:
Also, the test and gcsfuse logs are on different VM than Kokoro VM, so we will have to first copy to Kokoro VM to use that functionality. So, instead of first copying to Kokoro and then to bucket, we can directly copy to bucket.